First Release

n1kPLV opened the pull request:
Feedback on the service code in /Server/src/services/ would be great.

The code located in the folders /Website/ and /vehicle-simulator/ should be ready for review. However this is currently not the most important part of the codebase

NicoBiernat commented:
Temporarily closed such that the CI pipeline doesn't run twice on commits to development.
Please feel free to reopen this when necessary.

n1kPLV committed:
Added more a more specific exception to the fetcher

n1kPLV committed:
Preparations for tracker status display.

Also fixes #105

n1kPLV committed:
Started refactoring

n1kPLV committed:
Some error handling

n1kPLV committed:
Further refactorings and UI improvements

Commit:
#90: Adjusted Logging in track-service a little bit

Commit:
Merge branch 'development' into 91-mutating-array-functions-are-not-mutating

# Conflicts:
#	Server/src/services/poi.service.ts
#	Server/src/services/vehicle.service.ts

Commit:
Merge branch 'development' into 90-implement-logging-in-services

Kebslock committed:
Merge pull request #115 from kieler/91-mutating-array-functions-are-not-mutating

#91: Fixed mutation of certain array functions

Commit:
Merge branch 'development' into 90-implement-logging-in-services

Kebslock committed:
Merge pull request #119 from kieler/90-implement-logging-in-services

#90: Added logging to services

Commit:
Fixed init route

Commit:
Reordered methods in vehicle.route.ts

Commit:
Tiny adjustment to remove eslint violations

n1kPLV committed:
Hotfix for updating/deleting things with id 0

n1kPLV committed:
provide last known battery charge for each tracker

n1kPLV committed:
display tracker batery information (if known)

n1kPLV committed:
Small dark mode improvements

Commit:
#109: Refactor position computation and adjust time intervals

NicoBiernat committed:
Add README

NicoBiernat committed:
Fix eslint errors

Commit:
Merge branch 'development' into 109-adjust-time-interval-for-position-computation

Niatsuna committed:
Merge branch 'development' of github.com:kieler/RailTrail into 64-backend-refining-database-connection

JulianGrabitzky committed:
#65 install zod

JulianGrabitzky committed:
#65 Create and test UpdateTrack schema

Niatsuna committed:
Merge pull request #112 from kieler/64-backend-refining-database-connection

64 backend refining database connection

n1kPLV committed:
added pydantic for validating backups

Commit:
Merge branch 'development' into 109-adjust-time-interval-for-position-computation

Niatsuna committed:
Exclude Prisma directory

Niatsuna committed:
Add database seeding

JulianGrabitzky committed:
#129 Add first draft for multi-stage builds

Commit:
#109: Added fallback solution, where it was not used yet

n1kPLV committed:
further refactoring

Commit:
#109: Using vehicle speed computation for all position computations

jge-cap3 committed:
Change inital region

jge-cap3 committed:
Fix bottom layout to avoid blocking clicks

n1kPLV committed:
fixed infinite spinning on standalone track select

n1kPLV committed:
Merge pull request #117 from kieler/116-error-handling-in-management

116 error handling in management

n1kPLV committed:
Also provide vehicle speed on request

n1kPLV committed:
Add display of vehicle speed

Kebslock committed:
Merge pull request #124 from kieler/109-adjust-time-interval-for-position-computation

109 adjust time interval for position computation

n1kPLV committed:
Re-enable the default express exception handler

Commit:
#114: Replaced getNearbyVehicles

Commit:
Revert "#114: Replaced getNearbyVehicles"

This reverts commit 258510158d1451e7ee798c0b2105af5a85f2ec0c.

Commit:
#114: Replaced getNearbyVehicles

Commit:
added error handling concept

Commit:
Merge branch 'development' into 120-prisma-error-handling

Commit:
improved error handling

JulianGrabitzky committed:
Merge pull request #140 from kieler/114-remove-getnearbypois-vehicles

#114: Replaced getNearbyVehicles

Commit:
#133: Add conversion to percentage to track service

Commit:
#133: Replace getVehicleTrackDistancePercentage

Commit:
#133: Replaced getVehicleTrackDistanceKm

Kebslock committed:
Merge pull request #142 from kieler/133-remove-getvehicletrackdistancekm-and-move-getvehicletrackdistancepercentage

133 remove getvehicletrackdistancekm and move getvehicletrackdistancepercentage

n1kPLV committed:
Merge pull request #122 from kieler/tracker-components

Tracker status display

n1kPLV committed:
Updated package-lock

n1kPLV committed:
Restructured login process.

Fixes #135

Commit:
Improved null check of percentagePositions

Commit:
Improved null check of all occurrences of null checks with the ! operator

Commit:
little refactoring for services

Commit:
some refactoring to get rid of unsafe usages of the exclamation mark and some unused geojsonproperties

Commit:
removed unused imports from server.ts

n1kPLV committed:
Added Management of other users

Commit:
Made user deletion mechanism useful again

n1kPLV committed:
Merge pull request #145 from kieler/144-delete-other-users

Made user deletion mechanism useful again

Commit:
Added requested changes from review

n1kPLV committed:
Merge pull request #143 from kieler/101-pois-percentage-position

Null check on variables with type `number | null`

n1kPLV committed:
Added Management for the current user

n1kPLV committed:
Some final refactoring

n1kPLV committed:
Merge branch 'development' into 137-frontend-user-management

Commit:
#139: Enforcing order of computation for less redundancy

Commit:
Fixed last known vehicle position

Commit:
Merge branch 'development' into 139-backend-remove-redundant-calculations-and-database-queries

n1kPLV committed:
Merge pull request #146 from kieler/137-frontend-user-management

User management in the Frontend

n1kPLV committed:
Refactored map components

n1kPLV committed:
Focus map on track at start and zoom to fit track

n1kPLV committed:
Replaced redundant fetch functions

n1kPLV committed:
Added side-by-side map and list view

n1kPLV committed:
Translated more text into german

n1kPLV committed:
Handle long vehicle lists in advanced UI

n1kPLV committed:
Removed some debug logging

n1kPLV committed:
Fixed client exception occurring and implemented suggested changes from review

JulianGrabitzky committed:
#65 add zod schemas

n1kPLV committed:
Merge pull request #148 from kieler/136-frontend-side-by-side-map-list-interface

Side-by-side Map & List interface

closes #136
closes #138

JulianGrabitzky committed:
Merge branch 'development' into 65-payload-validation

JulianGrabitzky committed:
Merge branch 'development' into 65-payload-validation

JulianGrabitzky committed:
#65 Add validation to track route

JulianGrabitzky committed:
#65 remove schema validation from index route

JulianGrabitzky committed:
#65 add zod to init route

JulianGrabitzky committed:
Change first letter to uppercase

JulianGrabitzky committed:
#65 zod, login route

JulianGrabitzky committed:
#65 zod, poi route

JulianGrabitzky committed:
#65 zod, poitype route

JulianGrabitzky committed:
Convert api/tracker schema names to uppercase

JulianGrabitzky committed:
#65 zod, tracker route

JulianGrabitzky committed:
#65 zod, user route

JulianGrabitzky committed:
#65 zod, vehicle route

JulianGrabitzky committed:
#65 zod, vehicletype route

JulianGrabitzky committed:
#65 remove jsonschmea

JulianGrabitzky committed:
#65 adjust types in non-route files

JulianGrabitzky committed:
#65 adjust validation error severity in logging

JulianGrabitzky committed:
Adjust build step

JulianGrabitzky committed:
#65 remove finished todo's

Niatsuna committed:
Cascade Delete regarding POITypes and POIs

Niatsuna committed:
Soft Delete for Vehicles

Niatsuna committed:
Remove key pair for vehicles

Niatsuna committed:
Remove tracker ref for every inactive vehicle

Niatsuna committed:
Fix eslint mishaps

Niatsuna committed:
Soft Cascade Delete for VehicleTypes

Niatsuna committed:
Adjusted POITypes based on Feedback

Niatsuna committed:
Adjust creation of initial user based on feedback

Niatsuna committed:
Fix unqiue name mishap for vehicle types

n1kPLV committed:
Added crude restore functionality.

Will however fail if a poi type with the same name but different id is
present due to a uniqueness constraint.
Could probably be solved by matching names to existing poi types

Niatsuna committed:
Adjust error message for updateType

Niatsuna committed:
Fixed updateType

Niatsuna committed:
Add database Index for Logs

Niatsuna committed:
Adjust based on feedback

Niatsuna committed:
Merge pull request #150 from kieler/127-backend-seed-database

127 [Backend] Seeding of Database

Niatsuna committed:
Add getLatestLog

Niatsuna committed:
Adjust getAll with limit 1 to getLatestLog

Commit:
Implemented suggested changes of PR#147

Commit:
Merge branch 'development' into 139-backend-remove-redundant-calculations-and-database-queries

JulianGrabitzky committed:
Merge branch 'development' into 129-multi-stage-build

JulianGrabitzky committed:
#65

NicoBiernat committed:
Add endpoint for Oyster 3 LTE tracker

NicoBiernat committed:
Merge branch 'integrate-lte-tracker' into development

JulianGrabitzky committed:
Merge pull request #131 from kieler/129-multi-stage-build

#129 Add basic multi-stage builds

NicoBiernat committed:
Integrate LTE tracker, send correct status codes

NicoBiernat committed:
Merge branch 'development' of github.com:kieler/RailTrail into development

Niatsuna committed:
Merge pull request #151 from kieler/128-backend-soft-delete-cascading-deletes-for-datatypes

128 [Backend] Refinenment of Deletion for Vehicles, VehicleTypes and POIs

JulianGrabitzky committed:
Merge pull request #141 from kieler/120-prisma-error-handling

120 prisma error handling

JulianGrabitzky committed:
Merge branch 'development' into 65-payload-validation

JulianGrabitzky committed:
#65 add payload validation for lte payload

JulianGrabitzky committed:
Merge pull request #125 from kieler/65-payload-validation

Payload validation and parsing with zod

Commit:
Fixed JSDoc for getVehicleTrackHeading

Commit:
Merge branch 'development' into 139-backend-remove-redundant-calculations-and-database-queries

Commit:
Little fix after merging development

Kebslock committed:
Merge pull request #147 from kieler/139-backend-remove-redundant-calculations-and-database-queries

139 backend remove redundant calculations and database queries

n1kPLV commented:
Feedback on the service code in `/Server/src/services/` would be great.

The code located in the folders `/Website/` and `/vehicle-simulator/` should be ready for review. However this is currently not the most important part of the codebase

JulianGrabitzky committed:
Merge pull request #155 from kieler/app-main

Update App on development

JulianGrabitzky committed:
Merge branch 'development' into 152-backend-add-getlatest-to-logcontroller

JulianGrabitzky committed:
Merge pull request #156 from kieler/152-backend-add-getlatest-to-logcontroller

152 backend add getlatest to logcontroller

Commit:
changed poitypeicon enum in db seed

Commit:
Merge remote-tracking branch 'origin/development' into development

Commit:
Removed return await calls.

liamboddin committed:
Merge pull request #159 from kieler/158-return-await

Removed return await calls

Commit:
Added timestamp in backend code

Commit:
Changed API to better match the gps functionality in the phone

Commit:
Add null as viable trackerId

Commit:
Adjust findFirst and findUnique for Error handling

Commit:
Default error

Commit:
Adjust remove to void

Commit:
Merge branch '161-backend-remap-prisma-calls-for-v500' of github.com:kieler/RailTrail into 161-backend-remap-prisma-calls-for-v500

JulianGrabitzky committed:
Add input validation for init route

Commit:
Adjusted InitRoute

Commit:
Merge remote-tracking branch 'origin/161-backend-remap-prisma-calls-for-v500' into 161-backend-remap-prisma-calls-for-v500

Commit:
Adjusted VehicleTypeRoute

NicoBiernat committed:
Remove error handling for database from routes vehicletype, vehicle, tracker, poitype (handled in exception handler)

NicoBiernat committed:
Merge branch '161-backend-remap-prisma-calls-for-v500' of github.com:kieler/RailTrail into 161-backend-remap-prisma-calls-for-v500

Commit:
Fixed removes with no boolean return

JulianGrabitzky committed:
Remove null handeling for db service calls

JulianGrabitzky committed:
Remove comment

JulianGrabitzky committed:
DB/Vehicle, check if argument is undefined

JulianGrabitzky committed:
Adjust db vehicle controller

Commit:
#154: Seperated logic and DB-queries in vehicle service

Commit:
Fixed typo

Commit:
Fixed POITypeIcon enum check

Commit:
#154: Adjusted API calls

n1kPLV committed:
Removed database error handling in track route

n1kPLV committed:
Removed signup process

Commit:
Removed unnecessary call for trackers in app vehicle route

Commit:
Some restructuring

Commit:
Reordering in track route

Commit:
Removed whoAmI route

Commit:
Removed id from poitype api

Commit:
Fixed formatting in website code

n1kPLV committed:
Fixed battery level display

Commit:
Moved call from appendLog of app to trackerservice

NicoBiernat committed:
Add internal HTTPError that can be thrown by services

NicoBiernat committed:
Add type constraints to zod schema to avoid checking manually

NicoBiernat committed:
Improve error handling in user, login, tracker and crypto services, add expiry to JWTs

NicoBiernat committed:
Merge branch '161-backend-remap-prisma-calls-for-v500' of github.com:kieler/RailTrail into 161-backend-remap-prisma-calls-for-v500

n1kPLV committed:
Additional documentation (#166)

* Added some jsdoc comments

* Added even more JSDoc comments describing every function in this website

JulianGrabitzky committed:
Throw and catch zod errors

JulianGrabitzky committed:
Expose less with zod errors..

Commit:
#154: Changed returned position of getVehicleData for tracker logs only

Commit:
Merge branch 'development' into 154-compute-vehicle-data-all-at-once

NicoBiernat committed:
Remove unused imports (please install a linter so I don't have to be your garbage collector, thanks)

NicoBiernat committed:
Merge branch '154-compute-vehicle-data-all-at-once' into 161-backend-remap-prisma-calls-for-v500

NicoBiernat committed:
Catch error correctly after refactor

NicoBiernat committed:
Remove unnecessary null checks for functions that throw exceptions now

Kebslock committed:
Merge pull request #168 from kieler/154-compute-vehicle-data-all-at-once

154 compute vehicle data all at once

n1kPLV committed:
Improved error handling when getting multiple vehicles

NicoBiernat committed:
Catch exceptions instead of null-checking, remove including track

NicoBiernat committed:
Fix dereference of undefined if no tracker logs exist, also use tracker logs for computing heading and speed

NicoBiernat committed:
Log stack trace for internal server errors

NicoBiernat committed:
Merge branch '161-backend-remap-prisma-calls-for-v500' of github.com:kieler/RailTrail into 161-backend-remap-prisma-calls-for-v500

n1kPLV committed:
Merge pull request #157 from kieler/153-backend-indexes-for-logs-based-on-timestamps

Add database Index for Logs

NicoBiernat committed:
Merge pull request #163 from kieler/161-backend-remap-prisma-calls-for-v500

161 backend remap prisma calls for v500

Commit:
#169: Filtered app logs for less computation and better performance

Commit:
#170: Using HTTPError instead of returning null in GeoJSONUtils

Commit:
#170: Using HTTPError instead of returning null in services

Commit:
Merge branch '170-error-handling-in-services' into 169-fix-vehicle-data-computation

n1kPLV committed:
Fixed Frontend invalid token handling

Commit:
Merge branch 'development' into 160-add-timestamp

Commit:
Merge branch 'development' into 160-add-timestamp

# Conflicts:
#	Server/src/routes/vehicle.route.ts

Commit:
Fixed merging trouble

Commit:
Document the timestamp

Commit:
Fixed create user

Commit:
#170: Added JSDoc for throwing errors

Commit:
#170: Adapted API to changes of error handling in services

Commit:
Little fix for poi route

NiklasRentzCAU reviewed (commented):
We reviewed the service code in `/Server/src/services/`. The code structure is good, with comments to guide us through what each service does, the granularity and distinction between plain DB access and the services with logic is good. Some minor spelling and consistency errors in the comments are here and there, but nothing severe.

The logic and formulas could be discussed (e.g. the position calculation), but this seems to be yet incomplete in your internal talk about the calculation and from your documentation about this part.

We also expect that the general structure (use of the services as an interim to the DB/server-side handling of raw data) will be included in the Wiki in the future, currently dubbed "TODO".

Kebslock committed:
Merge pull request #171 from kieler/170-error-handling-in-services

170 error handling in services

Commit:
Merge branch 'development' into 169-fix-vehicle-data-computation

Commit:
#169: Fixed computation of vehicle's position

Commit:
#169: Fixed confusing error messages

Commit:
#169: Really fixed confusing error messages

Commit:
#118: Improved computation of vehicle heading and direction

JulianGrabitzky committed:
#165 Add real timestamp from payload to tracker logs

JulianGrabitzky committed:
Merge branch 'development' into 160-add-timestamp

JulianGrabitzky committed:
Merge pull request #172 from kieler/160-add-timestamp

160 add timestamp

JulianGrabitzky committed:
#162 rework poi db controller

n1kPLV committed:
Improved POI Type matching and restoration

Commit:
#169: Added requested changes

Commit:
Merge branch '169-fix-vehicle-data-computation' into 132-refactor-computation-of-vehicle-speed

n1kPLV committed:
Restructured Header

Commit:
#132: Improved computation of vehicle speed

n1kPLV committed:
Added smart-ish links to the header

n1kPLV committed:
Added a modal track selection dialog for most use cases

n1kPLV committed:
Fixed Layout

n1kPLV committed:
Improved some comments

Kebslock committed:
Merge pull request #173 from kieler/169-fix-vehicle-data-computation

169 fix vehicle data computation

Kebslock committed:
Merge pull request #174 from kieler/118-refactor-computation-for-vehicle-heading

118 refactor computation for vehicle heading

Kebslock committed:
Merge pull request #176 from kieler/132-refactor-computation-of-vehicle-speed

132 refactor computation of vehicle speed

Kebslock committed:
Merge pull request #177 from kieler/175-frontend-improve-user-navigation

Improve user navigation on the frontend

n1kPLV committed:
Fixed focus parameter for the map

Commit:
Mapping also last known vehicle position and heading on track

n1kPLV committed:
Merge pull request #123 from kieler/110-poi-backup-and-restore

110 poi backup and restore

n1kPLV committed:
Synchronized map position and zoom with React state

n1kPLV committed:
Resizing POI icons based on map zoom level

n1kPLV committed:
Merge pull request #178 from kieler/poi_icon_scaling

Poi icon scaling

n1kPLV committed:
Added timezone information in the vehicle simulator payload

n1kPLV committed:
Fixed heading calculation with only one tracker

n1kPLV committed:
updated docker-compose.yml with correct backend URI

n1kPLV committed:
Bumped Next.js version and removed an unused dependency

n1kPLV committed:
Fully removed an unused dependency

n1kPLV committed:
Fixed a type error

n1kPLV committed:
Added README for POI backup

Kebslock committed:
Update README.md

Kebslock committed:
Update README.md

Commit:
Fixed typo in readme

NicoBiernat committed:
Refactor deployment (just a few minor things)

a-sr commented:
Project finished
